library/Vanilla/Permissions.php

```diff
@@ -274,6 +293,21 @@ public function setBanned($isBanned) {
     }
 
     /**
+     * Get the flag to determine if all permission checks are being overridden.
+     *
+     * @param bool|null $overrideAll
+     * @return $this
+     */
+    public function setOverrideAll($overrideAll) {
+        if ($overrideAll !== null) {
+            $overrideAll = boolval($overrideAll);
```

Comment:
Vanilla makes use of PHP's non-boolean falsey values everywhere, I think this is a nice safeguard. One of the contributing factors of the bug that @DaazKu refers to was because a method returned a falsey value rather than a boolean. So, bugs can be introduced by assuming this is a boolean, when somebody may pass it another type. I agree with @initvector . Coercing a property to be the type it expects is a nice enhancement.